### DISCUSSION TITLE
Alamopy - surrogate modeling

### DISCUSSION THREAD

**Discussion by juanbaquero4** [IDAES/idaes-pse#1479] [Q&A] [answered]
Hi everybody! I need your help!

I am trying to use AlamoPy for surrogate modeling. but I am having trouble trying to generate it:

1.  How can I check if AlamoPy is installed in my Python environment? Is there a specific command or method to confirm its installation?

2. If AlamoPy is not installed, what is the recommended way to install it? Is this package included in IDAES package software installation?

The reason I am asking this is because I am trying to generate a surrogate modeling following the example is posted on github :

```
from io import StringIO
import sys

stream = StringIO()
oldstdout = sys.stdout
sys.stdout = stream

# Create ALAMO trainer object
trainer = AlamoTrainer(
    input_labels=input_cell_labels,
    output_labels=output_cell_labels,
    training_dataframe=data_training,
)

# Set ALAMO options
trainer.config.constant = True
trainer.config.linfcns = True
trainer.config.multi2power = [1, 2]
trainer.config.monomialpower = [2, 3]
trainer.config.ratiopower = [1, 2]
trainer.config.maxterms = [10] * len(output_cell_labels)  # max for each surrogate
trainer.config.filename = os.path.join(os.getcwd(), "alamo_run.alm")
trainer.config.overwrite_files = True
```

After running this for me is possible to check the file "alamo_run.alm", after running the following code this message error pops out:

`success, alm_surr, msg = trainer.train_surrogate() `

FileNotFoundError: Could not find ALAMO executable. Please ensure that ALAMO is installed and in the system path, or provide a path to the executable.

Immensely appreciate your help!

**Comment by andrewlee94** [comment, marked as answer]
@juanbaquero4 The issue is that ALAMOpy is just a Python interface to ALAMO, which is a separate commercial tool. The ALAMOpy interface is part of IDAES and so you will have it if you have installed IDAES (which I assume you have), but we do not (and cannot) distribute the actual ALAMO code with this as it requires a license.

You can get ALAMO from https://minlp.com/alamo-modeling-tool, and there is a 1-month trial license available ,or if you are at an educational institution I believe you can get a free academic license.

**Reply by juanbaquero4** [reply]
Thank you!!
